File: 1812/Domain/Game.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace _1812.Domain
{
    public class Game
    {
        public Player PlayerOne { get; private set; }
        public Player PlayerTwo { get; private set; }
        public Deck Deck { get; private set; }
        public HeadQuarter HQ { get; private set; }

        private int turn = 0;

        public Game(Player player1, Player player2, Deck deck, HeadQuarter hq, int turn)
        {
            PlayerOne = player1;
            PlayerTwo = player2;
            Deck = deck;
            HQ = hq;
            this.turn = turn;
        }

        private void Preparings()
        {
            List<Card> toPlace1 = Api.ChooseCardsToPlaceOnTablet(PlayerOne.hand);
            List<Card> toPlace2 = Api.ChooseCardsToPlaceOnTablet(PlayerTwo.hand);

            foreach (Card card in toPlace1)
            {
                if (card.Type == Card.CardType.Battalion)
                {
                    PlayerOne.tablet.SetBattalion(card);
                }
                if (card.Type == Card.CardType.Cannon)
                {
                    PlayerOne.tablet.SetCannon(card);
                }
                if (card.Type == Card.CardType.Operation)
                {
                    PlayerOne.tablet.SetOperation(card);
                }
            }
            foreach (Card card in toPlace2)
            {
                if (card.Type == Card.CardType.Battalion)
                {
                    PlayerTwo.tablet.SetBattalion(card);
                }
                if (card.Type == Card.CardType.Cannon)
                {
                    PlayerTwo.tablet.SetCannon(card);
                }
                if (card.Type == Card.CardType.Operation)
                {
                    PlayerTwo.tablet.SetOperation(card);
                }
            }

            turn += 1;
        }
    }
}

File: domain/game.py
from domain.player import Player
from domain.deck import Deck
from domain.head_quarter import HeadQuarter
from domain.api import Api
from domain.card import CardType

class Game:

    def __init__(self, player1: Player, player2: Player, deck: Deck, hq: HeadQuarter, turn: int):
        self.player1 = player1
        self.player2 = player2
        self.deck = deck
        self.hq = hq
        self.turn = turn


    def preparings(self):
        to_place1 = Api.choose_cards_to_place_on_tablet(self.player1.hand)
        to_place2 = Api.choose_cards_to_place_on_tablet(self.player2.hand)

        for card in to_place1:
            if card.type == CardType.Battalion:
                self.player1.tablet.set_battalion(card)

            if card.type == CardType.Cannon:
                self.player1.tablet.set_cannon(card)

            if card.type == CardType.Operation:
                self.player1.tablet.set_operation(card)
        
        for card in to_place2:
            if card.type == CardType.Battalion:
                self.player2.tablet.set_battalion(card)

            if card.type == CardType.Cannon:
                self.player2.tablet.set_cannon(card)

            if card.type == CardType.Operation:
                self.player2.tablet.set_operation(card)
            
        self.turn += 1
